File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/Format.java
package org.jvalue.ods.adapterservice.adapter;

import org.jvalue.ods.adapterservice.adapter.interpreter.*;

public enum Format {
  JSON(new JsonInterpreter()),
  XML(new XmlInterpreter()),
  CSV(new CsvInterpreter());

  private final Interpreter interpreter;
  Format(Interpreter interpreter) {
    this.interpreter = interpreter;
  }

  Interpreter getInterpreter() {
    return interpreter;
  }
}

File: adapter/src/adapter/Format.ts
import { CsvInterpreter } from './interpreter/CsvInterpreter';
import { Interpreter } from './interpreter/Interpreter';
import { JsonInterpreter } from './interpreter/JsonInterpreter';
import { XmlInterpreter } from './interpreter/XmlInterpreter';

export class Format {
  static readonly JSON = new JsonInterpreter();
  static readonly XML = new XmlInterpreter();
  static readonly CSV = new CsvInterpreter();

  interpreter: Interpreter;
  constructor(interpreter: Interpreter) {
    this.interpreter = interpreter;
  }

  getInterpreter(): Interpreter {
    return this.interpreter;
  }
}
